src/types/index-types.ts

```diff
@@ -0,0 +1,41 @@
+export type QueryOptions = {
+  sortProperty: string;
+  sortDirection?: SortDirection;
+  limit?: number;
+  cursor?: string;
+};
+
+export enum SortDirection {
+  Descending = -1,
+  Ascending = 1
+}
+
+export type EqualFilter = string | number | boolean;
+
+export type OneOfFilter = EqualFilter[];
+
+export type RangeValue = string | number;
+
+/**
+ * "greater than" or "greater than or equal to" range condition. `gt` and `gte` are mutually exclusive.
+ */
+export type GT = ({ gt: RangeValue } & { gte?: never }) | ({ gt?: never } & { gte: RangeValue });
+
+/**
+ * "less than" or "less than or equal to" range condition. `lt`, `lte` are mutually exclusive.
+ */
+export type LT = ({ lt: RangeValue } & { lte?: never }) | ({ lt?: never } & { lte: RangeValue });
+
+/**
+ * Ranger filter. 1 condition is required.
+ */
+export type RangeFilter = (GT | LT) & Partial<GT> & Partial<LT>;
+
+export type FilterValue = EqualFilter | OneOfFilter | RangeFilter;
+
+export type Filter = {
+  [property: string]: FilterValue;
+};
+
+export type Indexes = { [key: string]: unknown };
+export type IndexedItem<T> = { itemId: string, value: T, indexes: Indexes };
```

Comment:
Nit: love the new grouping of types, but not seeing the need to pull these two types out from `index-level.ts`, consider:
1. leaving these two types with `index-leverl.ts`
2. renaming the file to `query-types.ts`

Reply:
I was actually debating this exact thing as far as query-types vs index-types goes, but I'm fine with going with your suggestion and moving those two back into `index-level.ts`